common/src/main/java/com/wynntils/features/overlays/CustomBarsOverlayFeature.java

```diff
@@ -391,253 +145,156 @@ protected void renderBar(
-                            VerticalAlignment.Bottom,
-                            HorizontalAlignment.Center,
-                            OverlayPosition.AnchorSection.BottomMiddle),
-                    new OverlaySize(81, 21),
-                    CommonColors.YELLOW);
-        }
+    protected static class CustomBubbleBarOverlay extends CustomBarOverlayBase {
+        @RegisterConfig("overlay.wynntils.objectivesTexture")
+        public final Config<ObjectivesTextures> objectivesTexture = new Config<>(ObjectivesTextures.a);
 
-        @Override
-        public BossBarProgress progress() {
-            return Models.BossBar.focusBar.getBarProgress();
+        public CustomBubbleBarOverlay(int id) {
+            super(id, new OverlaySize(150, 30));
         }
 
         @Override
-        public String icon() {
-            return "➶";
+        public Texture getTexture() {
+            return Texture.BUBBLE_BAR;
         }
 
         @Override
-        public boolean isActive() {
-            return Models.BossBar.focusBar.isActive();
-        }
-    }
-
-    protected static class CorruptedBarOverlay extends BaseBarOverlay {
-        protected CorruptedBarOverlay() {
-            super(
-                    new OverlayPosition(
-                            -70,
-                            -150,
-                            VerticalAlignment.Bottom,
-                            HorizontalAlignment.Center,
-                            OverlayPosition.AnchorSection.BottomMiddle),
-                    new OverlaySize(81, 21),
-                    CommonColors.PURPLE);
+        protected float getTextureHeight() {
+            return 5;
         }
 
         @Override
-        public BossBarProgress progress() {
-            return Models.BossBar.corruptedBar.getBarProgress();
-        }
-
-        @Override
-        public String icon() {
-            return "☠";
+        protected void renderBar(
+                PoseStack poseStack,
+                MultiBufferSource.BufferSource bufferSource,
+                float renderY,
+                float barHeight,
+                float progress) {
+            BufferedRenderUtils.drawProgressBar(
+                    poseStack,
+                    bufferSource,
+                    getTexture(),
+                    getRenderX(),
+                    renderY,
+                    getRenderX() + getWidth(),
+                    renderY + barHeight,
+                    0,
+                    objectivesTexture.get().getTextureYOffset(),
+                    182,
+                    objectivesTexture.get().getTextureYOffset() + 10,
+                    progress);
         }
 
         @Override
-        public boolean isActive() {
-            return Models.BossBar.corruptedBar.isActive();
+        protected BarOverlayTemplatePair getActualPreviewTemplate() {
+            return new BarOverlayTemplatePair("{capped_ingredient_pouch_slots}", "capped_ingredient_pouch_slots");
```

Comment:
I'm not sure if I'm reading the diff incorrectly, but it looks like the BubbleBar is based on ingredient pouch slots?

Reply:
Yes, the preview is. I had to make a random example, let me know if I should change it.

Reply:
I now think the CustomBubbleBar is really like "CustomBarWithBubbleTexture". Is that a correct understanding? Then the example makes sense. But that was not really clear to me.

Reply:
Was it too tricky to have just a single type of CustomBar in which you could select the texture to use? Or are there other differences that I don't understand between the different kinds of custom bars?

Reply:
You got it right. And we need the different subclasses because health/mana/experience/bubble have different texture types themselves. And we have the universal as colorable, but a single texture.

Reply:
Let's say I hope for a future improvement in which we can select texture as well for a custom bar :-)

Maybe you can rename the classes a bit to make this clear, like `CustomBarWithBubbleTexture` or something like that?

Reply:
> Let's say I hope for a future improvement in which we can select texture as well for a custom bar :-)
> 
> Maybe you can rename the classes a bit to make this clear, like `CustomBarWithBubbleTexture` or something like that?

I am not quite sure `CustomBarWithBubbleTextureOverlay` sounds better. Or does it..?

Reply:
Also let me know what you think here.

Reply:
No, I don't think it sounds better. But I think the other name is confusing, like the bar did something *more* than just have a different texture. Especially for Health and Mana. As I said, I hope we can refactor this in the future so there is only one type of custom overlay with a selectable texture. Until then, I think it needs to be clear that the different subclasses only exists to provide different textures. Maybe `BubbleTexturedCustomBarOverlay` then?

Reply:
Renamed it!